SDL/Segment.cuh

```diff
@@ -324,9 +324,71 @@ namespace SDL
+        segmentsInGPU.dPhiChanges[idx]    = __F2H(dPhiChange);
+        segmentsInGPU.dPhiChangeMins[idx] = __F2H(dPhiChangeMin);
+        segmentsInGPU.dPhiChangeMaxs[idx] = __F2H(dPhiChangeMax);
+    }
 
-    ALPAKA_FN_ACC void addPixelSegmentToMemory(struct segments& segmentsInGPU, struct miniDoublets& mdsInGPU, struct modules& modulesInGPU, unsigned int innerMDIndex, unsigned int outerMDIndex, uint16_t pixelModuleIndex, unsigned int hitIdxs[4], unsigned int innerAnchorHitIndex, unsigned int outerAnchorHitIndex, float dPhiChange, float ptIn, float ptErr, float px, float py, float pz, float etaErr, float eta, float phi, int charge, unsigned int seedIdx, unsigned int idx, unsigned int pixelSegmentArrayIndex, int superbin, int8_t pixelType, short isQuad, float score);
+    template<typename TAcc>
+    ALPAKA_FN_ACC ALPAKA_FN_INLINE void addPixelSegmentToMemory(TAcc const & acc, struct segments& segmentsInGPU, struct miniDoublets& mdsInGPU, unsigned int innerMDIndex, unsigned int outerMDIndex, uint16_t pixelModuleIndex, unsigned int hitIdxs[4], unsigned int innerAnchorHitIndex, unsigned int outerAnchorHitIndex, float dPhiChange, unsigned int idx, unsigned int pixelSegmentArrayIndex, float score)
+    {
+        segmentsInGPU.mdIndices[idx * 2] = innerMDIndex;
+        segmentsInGPU.mdIndices[idx * 2 + 1] = outerMDIndex;
+        segmentsInGPU.innerLowerModuleIndices[idx] = pixelModuleIndex;
+        segmentsInGPU.outerLowerModuleIndices[idx] = pixelModuleIndex;
+        segmentsInGPU.innerMiniDoubletAnchorHitIndices[idx] = innerAnchorHitIndex;
+        segmentsInGPU.outerMiniDoubletAnchorHitIndices[idx] = outerAnchorHitIndex;
+        segmentsInGPU.dPhiChanges[idx] = __F2H(dPhiChange);
+        segmentsInGPU.isDup[pixelSegmentArrayIndex] = false;
+        segmentsInGPU.score[pixelSegmentArrayIndex] = score;
+
+        segmentsInGPU.pLSHitsIdxs[pixelSegmentArrayIndex].x = hitIdxs[0];
+        segmentsInGPU.pLSHitsIdxs[pixelSegmentArrayIndex].y = hitIdxs[1];
+        segmentsInGPU.pLSHitsIdxs[pixelSegmentArrayIndex].z = hitIdxs[2];
+        segmentsInGPU.pLSHitsIdxs[pixelSegmentArrayIndex].w = hitIdxs[3];
+
+        //computing circle parameters
+        /*
+        The two anchor hits are r3PCA and r3LH. p3PCA pt, eta, phi is hitIndex1 x, y, z
+        */
+        float circleRadius = mdsInGPU.outerX[innerMDIndex] / (2 * k2Rinv1GeVf);
+        float circlePhi = mdsInGPU.outerZ[innerMDIndex];
+        float candidateCenterXs[] = {mdsInGPU.anchorX[innerMDIndex] + circleRadius * alpaka::math::sin(acc, circlePhi), mdsInGPU.anchorX[innerMDIndex] - circleRadius * alpaka::math::sin(acc, circlePhi)};
+        float candidateCenterYs[] = {mdsInGPU.anchorY[innerMDIndex] - circleRadius * alpaka::math::cos(acc, circlePhi), mdsInGPU.anchorY[innerMDIndex] + circleRadius * alpaka::math::cos(acc, circlePhi)};
+
+        //check which of the circles can accommodate r3LH better (we won't get perfect agreement)
+        float bestChiSquared = 123456789.f;
```

Comment:
This should probably be `SDL_INF`?
...which should probably move to the `SDL/Constants.cu` file, instead of the `MiniDoublet.cu`?

Reply:
Sorry, I must have looked in a wrong branch? All good.